core/src/sumeragi/fault.rs

```diff
@@ -1305,500 +872,511 @@ pub fn run_sumeragi_main_loop<F>(
                             .expect("Sign genesis block.");
 
                         if !state_machine_guard.current_topology.is_consensus_required() {
-                            /*
-                            TODO: purge the unneeded of these
-                            self.invalidated_blocks_hashes.clear();
-                            self.txs_awaiting_created_block.clear();
-                            self.txs_awaiting_receipts.clear();
-                            self.votes_for_view_change.clear();
-                             */
-
                             sumeragi.broadcast_msg(
                                 BlockCommitted::from(signed_block.clone()),
                                 &state_machine_guard.current_topology,
                             );
 
-                            let block = signed_block.commit();
-                            let block_hash = block.hash();
-
-                            if let Err(error) = wsv_guard.apply(block.clone()) {
-                                panic!("Failed to apply block on WSV. This is absolutely not acceptable.");
-                            }
-
-                            for event in Vec::<Event>::from(&block) {
-                                trace!(?event);
-                                drop(sumeragi.events_sender.send(event));
-                            }
-
-                            state_machine_guard.latest_block_height = block.header().height;
-                            state_machine_guard.latest_block_hash = block.hash();
-
-                            let previous_role =
-                                state_machine_guard.current_topology.role(&sumeragi.peer_id);
-                            state_machine_guard.current_topology.apply_block(block_hash);
-                            info!(
-                                prev_peer_role = ?previous_role,
-                                new_peer_role = ?state_machine_guard.current_topology.role(&sumeragi.peer_id),
-                                new_block_height = %state_machine_guard.latest_block_height,
-                                %block_hash,
-                                "Committing block"
-                            );
-                            sumeragi.kura.blocking_store_block(block);
-                            SumeragiWithFault::<F>::update_network_topology(
-                                &mut state_machine_guard.current_topology,
-                                &wsv_guard,
-                            );
+                            block_commit(sumeragi, signed_block, &mut state_machine_guard);
                             has_sent_transactions = false;
                             voting_block_option = None;
+                            old_view_change_index = 0;
+                            view_change_proof_chain.clear();
```

Comment:
Maybe introduce function which would introduce state clean up?

Reply:
I tried that but didn't find it helped readability.